src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -782,29 +772,22 @@ public Thread(ThreadGroup group, Runnable target, String name,
      * @see        #stop()
      */
     public synchronized void start() {
-        /**
-         * This method is not invoked for the main method thread or "system"
-         * group threads created/set up by the VM. Any new functionality added
-         * to this method in the future may have to also be added to the VM.
-         *
-         * A zero status value corresponds to state "NEW".
-         */
-        if (threadStatus != 0)
-            throw new IllegalThreadStateException();
-
-        /* Notify the group that this thread is about to be started
-         * so that it can be added to the group's list of threads
-         * and the group's unstarted count can be decremented. */
-        group.add(this);
+        boolean success = false;
 
-        boolean started = false;
         try {
-            start0();
-            started = true;
+            synchronized (blockerLock) {
+                if (started) {
+                    throw new IllegalThreadStateException();
+                }
+
+                group.add(this);
+                startImpl();
+                success = true;
+            }
         } finally {
             try {
-                if (!started) {
-                    group.threadStartFailed(this);
+                if (!success && !started) {
+                    group.threadTerminated(this);
                 }
             } catch (Throwable ignore) {
```

Comment:
The comment in this `catch` block should be improved. The reference to `start0()` should be `startImpl()` and it should make it clearer that the throwable being ignored is from `ThreadGroup.threadStartFailed()`.

Reply:
Will updated the comment.